service/controller/resource/cloudconfig/desired.go

```diff
@@ -0,0 +1,164 @@
+package cloudconfig
+
+import (
+	"context"
+
+	"github.com/giantswarm/certs"
+	k8scloudconfig "github.com/giantswarm/k8scloudconfig/v6/pkg/template"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller/context/resourcecanceledcontext"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capzv1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capiv1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/exp/api/v1alpha3"
+	"sigs.k8s.io/cluster-api/util"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/cloudconfig"
+	"github.com/giantswarm/azure-operator/v4/service/controller/controllercontext"
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	cr, err := key.ToAzureMachinePool(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	machinePool, err := r.getOwnerMachinePool(ctx, cr.ObjectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	cluster, err := util.GetClusterFromMetadata(ctx, r.ctrlClient, cr.ObjectMeta)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	azureCluster, err := r.getAzureClusterFromCluster(ctx, cluster)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var clusterCerts certs.Cluster
+	{
+		clusterCerts, err = r.certsSearcher.SearchCluster(key.ClusterID(&cr))
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	storageAccountName := key.StorageAccountName(&cr)
+	containerName := key.BlobContainerName()
+	certificateEncryptionSecretName := key.CertificateEncryptionSecretName(&cr)
+
+	encrypter, err := r.toEncrypterObject(ctx, certificateEncryptionSecretName)
+	if apierrors.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "encryptionkey resource is not ready")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
+		return nil, nil
+	} else if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var ignitionTemplateData cloudconfig.IgnitionTemplateData
+	{
+		versions, err := k8scloudconfig.ExtractComponentVersions(cc.Release.Release.Spec.Components)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		defaultVersions := key.DefaultVersions()
+		versions.KubernetesAPIHealthz = defaultVersions.KubernetesAPIHealthz
+		versions.KubernetesNetworkSetupDocker = defaultVersions.KubernetesNetworkSetupDocker
+		images := k8scloudconfig.BuildImages(r.registryDomain, versions)
+
+		ignitionTemplateData = cloudconfig.IgnitionTemplateData{
+			AzureMachinePool: &cr,
```

Comment:
This should be the only difference between the old `blobobject` resource and this one.